src/wp-includes/theme-previews.php

```diff
@@ -54,3 +54,31 @@ function wp_attach_theme_preview_middleware() {
 		'after'
 	);
 }
+
+/**
+ * Set a JavaScript constant for theme activation.
+ *
+ * Sets the JavaScript global WP_BLOCK_THEME_ACTIVATE_NONCE containing the nonce
+ * required to activate a theme. For use within the site editor.
+ *
+ * @see https://github.com/WordPress/gutenberg/pull/41836.
+ *
+ * @since 6.3.0
+ * @private
+ */
+function wp_block_theme_activate_nonce() {
+	$nonce_handle = 'switch-theme_' . wp_get_theme_preview_path();
+	?>
+	<script type="text/javascript">
+		window.WP_BLOCK_THEME_ACTIVATE_NONCE = '<?php echo wp_create_nonce( $nonce_handle ); ?>';
```

Comment:
This could be a potential security issue. Is there a reason not to escape the nonce value?
```suggestion
		window.WP_BLOCK_THEME_ACTIVATE_NONCE = '<?php echo esc_js( wp_create_nonce( $nonce_handle ) ); ?>';
```

Reply:
Good catch @anton-vlasenko  -- it's more a bug than a security issue as it's not user input, however it may break with custom nonce implementations using special characters. 

It will need to change to `= <?php echo wp_json_encode( wp_create_nonce( $nonce_handle ) );` (without quotes) as `esc_js()` is intended for use in DOM attributes.

```
wp> esc_js( 'Pens & Pencils' );
=>  string(18) "Pens &amp; Pencils"
wp> wp_json_encode( 'Pens & Pencils' );
=> string(16) ""Pens & Pencils""
``` 

I'll reopen the ticket and add a PR to the GB repo.

Reply:
Thank you for getting back to my code review, @peterwilsoncc.

Yes, using `wp_json_encode()` is also an option.
In fact, I prefer `wp_json_encode()` over `esc_js()` because it automatically adds quotes.
I was just uncertain about which function aligns better with the "true WordPress way" of doing things.